odoo/addons/base/res/res_currency.py

```diff
@@ -80,6 +89,34 @@ def name_get(self):
         return [(currency.id, tools.ustr(currency.name)) for currency in self]
 
     @api.multi
+    def amount_to_text(self, amount):
+        self.ensure_one()
+        def _num2words(number, lang):
+            try:
+                return num2words(number, lang=lang).title()
+            except NotImplementedError:
+                return num2words(number, lang='en').title()
+
+        if num2words is None:
+            logging.getLogger(__name__).warning("The library 'num2words' is missing, cannot render textual amounts.")
+            return amount
+
+        fractional_value, integer_value = math.modf(amount)
```

Comment:
math.modf() will return the fractionnal and integer value with both having the sign of amount, so you need to call abs() on the fractionnal to avoid labelling twice the 'minus' in case of a negative number.
```
>>> import math
>>> math.modf(-25.32)
(-0.3200000000000003, -25.0)
```